ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/storage/StorageProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.storage;
+
+import tech.pegasys.pantheon.ethereum.db.BlockchainStorage;
+import tech.pegasys.pantheon.ethereum.mainnet.ProtocolSchedule;
+import tech.pegasys.pantheon.ethereum.worldstate.WorldStateStorage;
+
+import java.io.Closeable;
+
+public interface StorageProvider extends Closeable {
```

Comment:
Instead of our own new interface could we use a pair of javax.inject.Provider instances?  Do they need to be tied like this?  This would make applying a DI framework later much easier.  JSR-330 annotations have no tie to a specific DI framework.

https://docs.oracle.com/javaee/6/api/javax/inject/Provider.html

Reply:
The problem with having them supplied separately is that they will also have to be closed separately which is a problem because they will often use the same backend storage (RocksDB currently) and it's the actual storage backend that really gets closed.

Reply:
With a guice or dagger module we could aggregate the closables in the module.  But we don't have that in this CL so this interface will work.